be/src/olap/block_column_predicate.cpp

```diff
@@ -234,12 +234,16 @@ void AndBlockColumnPredicate::evaluate_vec(vectorized::MutableColumns& block, ui
         _block_column_predicate_vec[0]->evaluate_vec(block, size, flags);
     } else {
         bool new_flags[size];
+        bool initialized = false;
         for (auto block_column_predicate : _block_column_predicate_vec) {
-            memset(new_flags, true, size);
-            block_column_predicate->evaluate_vec(block, size, new_flags);
-
-            for (uint16_t j = 0; j < size; j++) {
-                flags[j] &= new_flags[j];
+            if (initialized) {
+                block_column_predicate->evaluate_vec(block, size, new_flags);
```

Comment:
If new flags is not init for every predicate, the value is not precise in new_flags for next predicate?